flink-runtime/src/main/java/org/apache/flink/runtime/scheduler/ExecutionGraphInfo.java

```diff
@@ -35,15 +33,15 @@
 
     private static final long serialVersionUID = -6134203195124124202L;
 
-    private final ArchivedExecutionGraph executionGraph;
-    private final List<ErrorInfo> exceptionHistory;
+    private final AccessExecutionGraph executionGraph;
```

Comment:
Is `ExecutionGraphInfo` still `Serializable` if it can contain an `AccessExecutionGraph` which is an unserializable `DefaultExecutionGraph`? Is [FileExecutionGraphInfoStore](https://github.com/XComp/flink/blob/e284963a18df3004b6b70fbcf38f1682c11d8584/flink-runtime/src/main/java/org/apache/flink/runtime/dispatcher/FileExecutionGraphInfoStore.java#L279) going to throw a  ` NotSerializableException` when it tries to serialize such an `ExecutionGraphInfo`?

Reply:
That's actually a good point I haven't considered when refactoring this piece of code. Theoretically it should not happen that a `DefaultExecutionGraph` is being returned by `ExecutionGraphInfo`. But it should reflected by the used interface as well. I'm gonna go ahead and introduce a `SerializableAccessExecutionGraph` @zentol do you see any problems with that approach?

Reply:
I suppose that would be an argument to stick with the ArchivedExecutionGraph instead.

Reply:
I don't see a lot of value in adding another interface for the sole purposes of differentiating between the 2 current implementations.

Reply:
I thought of keeping the interface considering that we might have to come up with something new for `AdaptiveScheduler`. But I'm also happy to revert the changes since we could argue that this would be out of scope for this ticket.

Reply:
Let's revert it for now and re-evaluate things down the line.